cpp/src/parquet/file_reader.cc

```diff
@@ -178,13 +179,15 @@ class SerializedRowGroup : public RowGroupReader::Contents {
                      std::shared_ptr<::arrow::io::internal::ReadRangeCache> cached_source,
                      int64_t source_size, FileMetaData* file_metadata,
                      int row_group_number, const ReaderProperties& props,
+                     const std::unordered_set<int>& prebuffered_column_chunks,
```

Comment:
nit: take this by move and move in the constructor? (so that the caller chooses whether to copy or not)